test.sh.in

```diff
@@ -59,18 +69,41 @@ dlTftpFiles() {
 
 prepareDeps() {
   rm -rf ${WORKSPACE}/build-deps
-  mkdir -p ${WORKSPACE}/build-deps/${REPO_NAME} 
+  if [ -z "${MULTI}" ]; then
+    mkdir -p ${WORKSPACE}/build-deps/${REPO_NAME} 
+  else
+    mkdir -p ${WORKSPACE}/build-deps
+    mkdir -p ${WORKSPACE}/build
+  fi
   for i in ${REPOS[@]}; do
      cd ${WORKSPACE}/build-deps
-     if [ "${i}" != "${REPO_NAME}" ]; then
-         git clone https://github.com/RackHD/${i}.git
+     if [[ "${REPO_NAME}" != *"${i}"* ]]; then
+         git clone https://github.com/changev/${i}.git
          cd ${i}
          git checkout ${GIT_REFSPEC}
          npm install --production
      fi
   done
   dlTftpFiles
   dlHttpFiles
+  apiPackageModify
+}
+
+apiPackageModify() {
+  if [[ "${REPO_NAME}" == *"on-http"* ]] && [[ "${REPO_NAME}" == *"rackhd"* ]]; then
+    pushd ${WORKSPACE}/on-http/extra
+    sed -i "s/.*git symbolic-ref.*/ continue/g" make-deb.sh
+    sed -i "/build-package.bash/d" make-deb.sh
+    sed -i "/mkdir/d" make-deb.sh
+    sudo bash make-deb.sh
+    popd
+    for package in ${API_PACKAGE_LIST}; do
+      pip uninstall -y ${package//./-}
+      pushd ${WORKSPACE}/on-http/$package
+        sudo python setup.py install
+      popd
+    done
+  fi
```

Comment:
When on-http updates, the python api packages which RackHD smoke-tests depends on may change too. 
The used solution is updating packages to pypi in travis-ci when a pr merged to on-http master.
Now when found on-http and rackhd both exists in REPO_NAME, we'll use local generated package to replace pypi package.